src/libraries/System.Data.OleDb/src/UnsafeNativeMethods.COMWrappers.cs

```diff
@@ -0,0 +1,19 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Runtime.InteropServices;
+
+namespace System.Data.Common
+{
+    internal static partial class UnsafeNativeMethods
+    {
+        //
+        // Oleaut32
+        //
+
+        [DllImport(Interop.Libraries.OleAut32, CharSet = CharSet.Unicode, PreserveSig = true)]
```

Comment:
I don't think we need `PreserveSig` on the "new" methods. We should just match the native signatures in a blittable way.

Reply:
You can also drop the `CharSet = CharSet.Unicode`

Reply:
Isn't 
```
[DllImport(Interop.Libraries.OleAut32)]
internal static extern System.Data.OleDb.OleDbHResult GetErrorInfo(int dwReserved, System.IntPtr* ppIErrorInfo);
```
become 
```
HRESULT GetErrorInfo(int dwReserved, int* ppIErrorInfo, *int retVal); 
```
?

Reply:
Nope. The P/Invoke signature for [`GetErrorInfo`](https://docs.microsoft.com/windows/win32/api/oleauto/nf-oleauto-geterrorinfo) should be:

```csharp
[DllImport(Interop.Libraries.OleAut32)]
internal static extern System.Data.OleDb.OleDbHResult GetErrorInfo(
    int dwReserved,
    System.IntPtr* ppIErrorInfo);
```

This is assuming `System.Data.OleDb.OleDbHResult` is an `enum` and not a `struct`.

Reply:
Yeah. I completely forgot that `PreserveSig=true` is default.